File: QE-Selenium-Java/Activity17.java
package demos;

import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.firefox.FirefoxDriver;
import org.openqa.selenium.support.ui.Select;

public class Activity17 {

    public static void main(String[] args) {

        WebDriver driver = new FirefoxDriver();

        try {
            driver.get("https://training-support.net/webelements/selects");
            System.out.println("Page Title: " + driver.getTitle());

            WebElement multiSelect = driver.findElement(
                    By.xpath("//select[@multiple]"));

            Select select = new Select(multiSelect);

            if (select.isMultiple()) {
                System.out.println("Multi-select dropdown detected");
            }

            select.selectByVisibleText("HTML");

            select.selectByIndex(3); // Node
            select.selectByIndex(4); // React
            select.selectByIndex(5); // Angular

            select.selectByValue("nodejs");

            Thread.sleep(2000);

            select.deselectByIndex(4);

            Thread.sleep(2000);

        } catch (Exception e) {
            e.printStackTrace();
        } finally {
            driver.quit();
        }
    }
}

File: QE-Selenium-Python/Activity17.py
from selenium import webdriver
from selenium.webdriver.common.by import By
from selenium.webdriver.support.ui import Select
import time

driver = webdriver.Firefox()

try:
    driver.get("https://training-support.net/webelements/selects")
    print("Page Title:", driver.title)

    multi_select = driver.find_element(By.XPATH, "//select[@multiple]")
    select = Select(multi_select)

    if select.is_multiple:
        print("Multi-select dropdown detected")

    select.select_by_visible_text("HTML")

    select.select_by_index(3)  # Node
    select.select_by_index(4)  # React
    select.select_by_index(5)  # Angular

    select.select_by_value("nodejs")

    time.sleep(2)

    select.deselect_by_index(4)

    time.sleep(2)

finally:
    driver.quit()
